splitio/models/grammar/matchers/rule_based_segment.py

```diff
@@ -32,11 +32,14 @@ def _match(self, key, attributes=None, context=None):
         # Check if rbs segment has exclusions
         if context['ec'].segment_rbs_memberships.get(self._rbs_segment_name):
             return False
-                
-        for parsed_condition in context['ec'].segment_rbs_conditions.get(self._rbs_segment_name):
-            if parsed_condition.matches(key, attributes, context):
+        
+        for rbs_segment in context['ec'].excluded_rbs_segments:
+            if self._match_conditions(rbs_segment.conditions, key, attributes, context):
                 return True
```

Comment:
this should be return false

Reply:
The function already return false. We don't pass the storage objects to matchers in Python, so async and thread base will use the same calls.